addons/l10n_id_efaktur_coretax/models/account_move.py

```diff
@@ -0,0 +1,330 @@
+        ('TD.01103', '3 - PPN dibebaskan berdasarkan Peraturan Pemerintah Nomor 28 Tahun 2009'),
+        ('TD.01104', '4 - (Tidak ada cap)'),
+        ('TD.01105', '5 - PPN Dibebaskan Sesuai Dengan PP Nomor 81 Tahun 2015'),
+        ('TD.01106', '6 - PPN Dibebaskan Berdasarkan PP Nomor 74 Tahun 2015'),
+        ('TD.01107', '7 - (tanpa cap)'),
+        ('TD.01108', '8 - PPN DIBEBASKAN SESUAI PP NOMOR 81 TAHUN 2015 SEBAGAIMANA TELAH DIUBAH DENGAN PP 48 TAHUN 2020'),
+        ('TD.01109', '9 - PPN DIBEBASKAN BERDASARKAN PP NOMOR 47 TAHUN 2020'),
+        ('TD.01110', '10 - PPN Dibebaskan berdasarkan PP Nomor 49 Tahun 2022')],
+        compute="_compute_l10n_id_coretax_facility_info",
+        readonly=False,
+        store=True,
+    )
+
+    l10n_id_kode_transaksi = fields.Selection(selection_add=[('10', '10 Other deliveries')])
+    l10n_id_coretax_efaktur_available = fields.Boolean(compute="_compute_l10n_id_coretax_efaktur_available")
+    l10n_id_coretax_document = fields.Many2one('l10n_id_efaktur_coretax.document', readonly=True, copy=False, string="e-Faktur Document")
+    l10n_id_coretax_custom_doc = fields.Char(help="Additional documentation when choosing kode 07 or 08", default="")
+
+    def _compute_need_kode_transaksi(self):
+        """ OVERRIDE: l10n_id_efaktur
+
+        By setting this l10n_id_need_kode_transaksi, we can prevent the old E-Faktur flow to be
+        triggered(i.e. efaktur range consumption).
+        """
+        self.l10n_id_need_kode_transaksi = False
+
+    @api.depends('partner_id', 'line_ids.tax_ids')
+    def _compute_l10n_id_coretax_efaktur_available(self):
+        """ Similar use case as l10n_id_need_kode_transaksi from l10n_id_efaktur
+
+        helps to check whether or not some fields need to be visible or not
+        """
+        for move in self:
+            move.l10n_id_coretax_efaktur_available = (
+                move.partner_id.l10n_id_pkp
+                and move.country_code == 'ID'
+                and move.move_type == 'out_invoice'
+                and move.line_ids.tax_ids
+            )
+
+    # Interactions that is triggered when choosing kode 07 or 08
+    # as they require additional information
+    @api.depends("l10n_id_coretax_add_info_07", "l10n_id_coretax_add_info_08")
+    def _compute_l10n_id_coretax_facility_info(self):
+        for move in self:
+            if move.l10n_id_kode_transaksi == "07":
+                digit = move.l10n_id_coretax_add_info_07
+                field = "l10n_id_coretax_facility_info_07"
+            elif move.l10n_id_kode_transaksi == "08":
+                digit = move.l10n_id_coretax_add_info_08
+                field = "l10n_id_coretax_facility_info_08"
+            else:
+                digit = False
+                field = False
+
+            if field:
+                if digit:
+                    move[field] = f"TD.011{digit[-2:]}"
+                else:
+                    move[field] = False
```

Comment:
```suggestion
            if move.l10n_id_kode_transaksi == "07":
                digits = move.l10n_id_coretax_add_info_07
                move.l10n_id_coretax_facility_info_07 = f"TD.011{digits[-2:]}"
            elif move.l10n_id_kode_transaksi == "08":
                digits = move.l10n_id_coretax_add_info_08
                move.l10n_id_coretax_facility_info_08 = f"TD.011{digits[-2:]}"
```
This is shorter, and it's easier to understand. Same below.

Reply:
but wouldn't there be possibility of "l10n_id_coretax_add_info.." to have null (False) value?

Reply:
I didn't say my example was complete :see_no_evil: 
Indeed you should still check if the add_info has been provided.

Reply:
ah I see, noted